### DISCUSSION TITLE
Automatic identification of rings in sisl.geometry()

### DISCUSSION THREAD

**Discussion by ialcon** [zerothi/sisl#685] [Ideas]
Hi there,

I just would like to open a discussion on the possibility of sisl to automatically identify rings in a given molecular structure or material. The most relevant case for this would be, for example, getting all hexagonal rings within a graphene nanoribbon (GNR) or nanoporous graphene (NPG). I guess the hexagons could be either created as new geometry objects or simply as lists of atomic labels, or something like that. Getting such ring objects/lists would then allow to obtain other properties from the input geometry, or to further classify it (for example, identifying the two separate GNRs within an NPG structure). 

I am not sure if such type of a feature is already implemented in SISL. However, if that is not the case, and you find such an implementation would be generally useful, I can share some code that I used myself to search for aryl rings in 1D conjugated polymer geometries.

Looking for your opinion!

Isaac

**Comment by zerothi** [comment]
Thanks for reaching out! 

Currently there is no such feature, but indeed there seems to should be.
There are various ways to use the infrastructure here, and I presume that if we could use the `GeometryCategory` infrastructure to accommodate something like this, it would be useful.  
Having categories that stores indices of atoms in the category could be useful.

Sharing your code would be super useful to discover the possibilities etc.

**Reply by ialcon** [reply]
Ok, Nick! Now I have to run, but tomorrow anytime I will share the parts of my code that are related to this issue. Should I share them directly in this discussion, or create an "issue" for this?

**Reply by zerothi** [reply]
Probably it would be best to create an issue, then we can discuss a bit, then we can always move it to a PR, thanks!
